pkg/codegen/python/gen.go

```diff
@@ -513,8 +514,51 @@ def get_version():
      return _version_str
 `)
 	}
+	if err != nil {
+		return nil, err
+	}
+
+	if pkg.Parameterization != nil {
+		// If a parameterized package is being generated then we _need_ to use package references
+		param := base64.StdEncoding.EncodeToString(pkg.Parameterization.Parameter)
+
+		_, err = fmt.Fprintf(buffer, `
+_package_lock = asyncio.Lock()
+_package_ref = ...
+async def get_package():
+	global _package_ref
+	if _package_ref is ...:
+		async with _package_lock:
+			if _package_ref is ...:
+				monitor = pulumi.runtime.settings.get_monitor()
+				parameterization = resource_pb2.Parameterization(
+					name=%q,
+					version=get_version(),
+					value=base64.b64decode(%q),
+				)
+				try:
+					registerPackageResponse = monitor.RegisterPackage(
+						resource_pb2.RegisterPackageRequest(
+							name=%q,
+							version=%q,
+							download_url=get_plugin_download_url(),
+							parameterization=parameterization,
+						))
+					_package_ref = registerPackageResponse.ref
+				except Exception as e:
+					raise e
+					_package_ref = None
```

Comment:
If you `raise` first, will you hit the line where you assign `None`?

Reply:
Ah the whole try/except should go really, before we stable ship this we'll add a feature flag for support and then if RegisterPackage fails we want that to bubble to the user.